src/generalVis/components/GeneralHome.tsx

```diff
@@ -0,0 +1,265 @@
+    const [opacity, setOpacity] = useState<NumericalColumn | null>(null);
+    const [shape, setShape] = useState<CategoricalColumn | null>(null);
+    const [barGroup, setBarGroup] = useState<CategoricalColumn | null>(null);
+    const [barMultiples, setBarMultiples] = useState<CategoricalColumn | null>(null);
+    const [barDisplayType, setBarDisplayType] = useState<EBarDisplayType>(EBarDisplayType.DEFAULT);
+    const [barGroupType, setBarGroupType] = useState<EBarGroupingType>(EBarGroupingType.STACK);
+    const [barDirection, setBarDirection] = useState<EBarDirection>(EBarDirection.VERTICAL);
+
+    const updateBubbleSize = (newCol: string) => setBubbleSize(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.NUMERICAL)[0] as NumericalColumn);
+    const updateOpacity = (newCol: string) => setOpacity(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.NUMERICAL)[0] as NumericalColumn);
+    const updateColor = (newCol: string) => setColorMapping(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+    const updateShape = (newCol: string) => setShape(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+    const updateBarGroup = (newCol: string) => setBarGroup(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+    const updateBarMultiples = (newCol: string) => setBarMultiples(props.columns.filter((c) => c.name === newCol && c.type === EColumnTypes.CATEGORICAL)[0] as CategoricalColumn);
+
+    const updateCurrentVis = (s: ESupportedPlotlyVis) => setCurrentVis(s);
+    const updateSelectedCatCols = (s: string[]) => setSelectedCatCols(s);
+    const updateSelectedNumCols = (s: string[]) => setSelectedNumCols(s);
+
+    const updateBarDisplayType = (s: EBarDisplayType) => setBarDisplayType(s);
+    const updateBarGroupType = (s: EBarGroupingType) => setBarGroupType(s);
+    const updateBarDirection = (s: EBarDirection) => setBarDirection(s);
+
+    const shapeScale = useMemo(() => {
+        return shape ?
+            scale.ordinal<string>().domain(d3.set(shape.vals.map((v) => v.val)).values()).range(['circle', 'square', 'triangle-up', 'star'])
+            : null;
+    }, [shape]);
+
+    const bubbleScale = useMemo(() => {
+        return bubbleSize ?
+            scale.linear().domain([0, d3.max(bubbleSize.vals.map((v) => v.val))]).range([0, 10])
+            : null;
+    }, [bubbleSize]);
+
+    const opacityScale = useMemo(() => {
+        return opacity ?
+            scale.linear().domain([0, d3.max(opacity.vals.map((v) => v.val))]).range([0, 1])
+            : null;
+    }, [opacity]);
+
+    const colorScale = useMemo(() => {
+        return scale.ordinal().range(['#337ab7', '#ec6836', '#75c4c2', '#e9d36c', '#24b466', '#e891ae', '#db933c', '#b08aa6', '#8a6044', '#7b7b7b']);
```

Comment:
@dvdanielamoitzi will add the color scale to phovea_ui. Once available, please repelace this array here.